clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -225,11 +230,20 @@ public double compressionRate() {
     }
 
     public boolean isFull() {
-        return recordsBuilder.isFull();
+        return !isWritable || recordsBuilder.isFull();
```

Comment:
Not sure if we strictly needs to isWriteable. Currently, during append(), if the current producerBatch is full, we just create a new batch.